### DISCUSSION TITLE
How to handle exceptions from `emscripten_run_script`?

### DISCUSSION THREAD

**Discussion by tuket** [emscripten-core/emscripten#17225] [Q&A] [answered]
Hi,
I'm using `emscripten_run_script` to run some arbitrary JS code.
I can't guarantee that the JS code is correct.

Right now when there is an error in the JS code I get an exception, and the application can't continue. I would like to avoid that; the application should log the error and continue.

I have tried different things:
 - Wrapping the JS code with a try-catch `"try {" + str + "} catch(err) { console.log(err.message); }"`
 - Do the try-catch in C++: `try {emscripten_run_script(str);} catch (...) {};`
 - Add `-fexceptions` flag to compiler and linker

But none of then seem to work. What should I do?

Thanks!

**Comment by kripken** [comment, marked as answer]
The first option should work, if you do it on the JS that is sent to `emscripten_run_script`. If not, perhaps there is a typo in the try-catch or elsewhere? (that would throw before it runs the code) Or perhaps it throws in the catch body? (do all thrown `err` have the property `err.message`..?)

**Reply by tuket** [reply]
You are right, after doing a clean recompile it worked. Thanks!
